conversion/resource_generation.go

```diff
@@ -0,0 +1,413 @@
+// Sets Profile for resource creation
+func (r ResourceGenerationStruct) setConnectionProfileFromRequest(details *ConnectionProfileReq, req *datastreampb.CreateConnectionProfileRequest) error {
+	if details.ConnectionProfile.IsSource {
+		port, err := strconv.ParseInt((details.ConnectionProfile.Port), 10, 32)
+		if err != nil {
+			return err
+		}
+		req.ConnectionProfile.Profile = &datastreampb.ConnectionProfile_MysqlProfile{
+			MysqlProfile: &datastreampb.MysqlProfile{
+				Hostname: details.ConnectionProfile.Host,
+				Port:     int32(port),
+				Username: details.ConnectionProfile.User,
+				Password: details.ConnectionProfile.Password,
+			},
+		}
+	} else {
+		req.ConnectionProfile.Profile = &datastreampb.ConnectionProfile_GcsProfile{
+			GcsProfile: &datastreampb.GcsProfile{
+				Bucket:   details.ConnectionProfile.BucketName,
+				RootPath: "/",
+			},
+		}
+		return nil
+	}
+	return nil
+}
+
+func (r ResourceGenerationStruct) GetSpannerRegion(ctx context.Context, projectId string, instanceName string) (string, error) {
+	instanceAdmin, _ := instance.NewInstanceAdminClient(ctx)
+	defer instanceAdmin.Close()
+	region := ""
+	spannerInstance, err := instanceAdmin.GetInstance(ctx, &instancepb.GetInstanceRequest{Name: "projects/" + projectId + "/instances/" + instanceName})
+	if err != nil {
+		return region, err
+	}
+	instanceConfig, err := instanceAdmin.GetInstanceConfig(ctx, &instancepb.GetInstanceConfigRequest{Name: spannerInstance.Config})
+	if err != nil {
+		return region, err
+	}
+	for _, replica := range instanceConfig.Replicas {
+		if replica.DefaultLeaderLocation {
+			region = replica.Location
+		}
+	}
+	return region, nil
+}
+
+// Returns connection profiles for a given region
+func (r *ResourceGenerationStruct) getConnProfilesRegion(ctx context.Context, projectId string, region string, dsClient *datastream.Client) ([]string, error) {
+	profilesIt := dsClient.ListConnectionProfiles(ctx, &datastreampb.ListConnectionProfilesRequest{Parent: "projects/" + projectId + "/locations/" + region})
+	var profiles []string = []string{}
+	for {
+		resp, err := profilesIt.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return profiles, err
+		} else {
+			profiles = append(profiles, strings.Split(resp.Name, "/")[5])
+		}
+	}
+	return profiles, nil
+}
+
+// returns true if connection profile exists else false
+func (r *ResourceGenerationStruct) connectionProfileExists(ctx context.Context, projectId string, profileName string, profileLocation string, connectionProfiles map[string][]string, dsClient *datastream.Client, s resourceGenerationInterface) (bool, error) {
+	// Check if connection profiles for the given region are fetched. if not, fetch them
+	profiles, ok := connectionProfiles[profileLocation]
+	var err error = nil
+	if !ok {
+		profiles, err = s.getConnProfilesRegion(ctx, projectId, profileLocation, dsClient)
+		if err != nil {
+			return false, err
+		}
+		connectionProfiles[profileLocation] = profiles
+	}
```

Comment:
Looks like some of this code already be implemented when the UI feature of "listing connection profiles" was implemented. We should reuse/refactor code so that both flows use the same logic for common operations such as listing connection profiles from Datastream.

Reply:
It is, it'll get removed and this function will be called in the UI PR